### PR TITLE
Feature/leaf 1451 improve importer reliablity

### PR DESCRIPTION
**Pull request by g-theriot**

### CODE REVIEW

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl [hunk 1]
```diff
@@ -749,23 +862,44 @@
...
+
             }
 
             /* iterate through the sheet cells, which are organized by row */
-            for (var i = 1; i <= sheet_data.cells.length - 1; i++) {
-                selectRowToAnswer(i);
+            totalRecords = sheet_data.cells.length -1;
+            
+            dialog.dialog( "open" );
+            debugger;
```
Comment (jeremyrl) [inline LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl:888]: remove debugger

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl [hunk 2]
```diff
@@ -387,6 +417,54 @@
...
+
+        function closeImport() {
+            $("#modal-background").removeClass("modalBackground");
+            clearTimeout( progressTimer );
+            dialog
+                .dialog( "close" );
+            progressbar.progressbar( "value", false );
+            progressLabel
+                .text( "Starting import..." );
+        }
```
Comment (jeremyrl) [inline LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl:455]: Weird line breaks

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl [hunk 3]
```diff
@@ -592,22 +673,46 @@
...
+                                var addAnother = i+1 <= sheet_data.cells.length - 1;
+                                
+                                if(addAnother){
+                                    doublet.push(selectRowToAnswer(i+1));
+                                }
+                                console.log("Before: "+ totalImported);
+                                
+                                console.log(totalImported);
+                                Promise.all(doublet).then(function(){
+                                    console.log("over here");
```
Comment (jeremyrl) [inline LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl:713]: Remove console.log on 713 and 709

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl [hunk 4]
```diff
@@ -749,23 +862,44 @@
...
+                var doublet = [];
+                doublet.push(selectRowToAnswer(i));
+                
+                var addAnother = i+1 <= sheet_data.cells.length - 1;
+                
+                if(addAnother){
+                    doublet.push(selectRowToAnswer(i+1));
+                }
+                 Promise.all(doublet).then(function(){
+                     console.log("here");
```
Comment (jeremyrl) [inline LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl:900]: remove console.log

**Review by jeremyrl — CHANGES_REQUESTED** [review]

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl (hunk 1 above)
Comment (g-theriot) [reply LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl]: fixed

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl (hunk 4 above)
Comment (g-theriot) [reply LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl]: done

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl (hunk 2 above)
Comment (g-theriot) [reply LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl:454]: How dare you.  Fixed.

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl [hunk 5]
```diff
@@ -14,10 +14,26 @@
...
+        margin-top: 0px;
+        margin-left: 0px;
+        opacity: 0.5;
+    }
+
 </style>
 <script type="text/javascript" src="https://cdn.jsdelivr.net/gh/SheetJS/js-xlsx@1eb1ec/dist/xlsx.full.min.js"></script>
 <script type="text/javascript" src="https://cdn.jsdelivr.net/gh/SheetJS/js-xlsx@64798fd/shim.js"></script>
 <script type="text/javascript" src="js/lz-string/lz-string.min.js"></script>
+<link rel="stylesheet" href="https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css">
```
Comment (thomasnguyen95) [inline LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl:33]: can we just grab the files here so whenever this importer script is used there isnt an additional script fetch.

Comment (dastan-rahimi) [reply LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl:33]: Please add these files under libs folder

**Review by abiola-adewuyi — APPROVED** [review]

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl (hunk 5 above)
Comment (g-theriot) [reply LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl]: Done

**Pull request merged**